File: app/utils/user_data.lua
local ____exports = {}
local file_key = "sys_save_load"
function ____exports.set_data(key, tab)
    local filename = sys.get_save_file(file_key, key)
    sys.save(filename, tab)
end
function ____exports.get_data(key)
    local filename = sys.get_save_file(file_key, key)
    local data = sys.load(filename)
    local nx = next(data)
    if nx == nil then
        return nil
    end
    return data
end
function ____exports.get(key)
    print(key)
    local data = ____exports.get_data(key)
    if data == nil then
        return nil
    else
        return data.value
    end
end
function ____exports.set(key, val)
    ____exports.set_data(key, {value = val})
end
function ____exports.get_int(key, def)
    if def == nil then
        def = 0
    end
    local data = ____exports.get_data(key)
    if data == nil then
        return def
    end
    return data.value
end
function ____exports.get_bool(key, def)
    if def == nil then
        def = false
    end
    local val = ____exports.get_int(key, -1)
    if val == -1 then
        return def
    end
    return val == true
end
function ____exports.get_string(key, def)
    if def == nil then
        def = ""
    end
    local data = ____exports.get_data(key)
    if data == nil then
        return def
    end
    return data.value or ""
end
return ____exports

File: src/utils/user_data.ts
/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
const file_key = "sys_save_load";


export function set_data(key: string, tab: any) {
    const filename = sys.get_save_file(file_key, key);
    sys.save(filename, tab);
}

export function get_data(key: string) {
    const filename = sys.get_save_file(file_key, key);
    const data = sys.load(filename);
    const [nx] = next(data);
    if (nx == null)
        return null;
    return data;
}

export function get(key: string) {
    print(key);
    const data = get_data(key);
    if (data == null)
        return null;
    else
        return data.value;
}



export function set(key: string, val: any) {
    set_data(key, { value: val });
}

export function get_int(key: string, def = 0): number {
    const data = get_data(key);
    if (data == null)
        return def;
    return data.value;
}

export function get_bool(key: string, def = false): boolean {
    const val: any = get_int(key, -1);
    if (val == -1)
        return def;
    return val == true;
}

export function get_string(key: string, def = ''): string {
    const data = get_data(key);
    if (data == null)
        return def;
    return data.value || '';
}

